File: src/test/kotlin/com/example/sandbox/concurrent/ExecutorTest.java
package com.example.sandbox.concurrent;

import org.junit.jupiter.api.Test;

import java.util.concurrent.*;

public class ExecutorTest {

    @Test
    void test1() {
        Executor executor = new ForkJoinPool();

        executor.execute(new SomeRun());
    }

    @Test
    void test2() {
        Executor executor = new ThreadPoolExecutor(5, 10, 10000L, TimeUnit.MINUTES, new LinkedBlockingQueue());

        for (int i = 0; i < 20; i++) {
            executor.execute(new SomeRun());
        }
    }

    class SomeRun implements Runnable {
        @Override
        public void run() {
            System.out.println("it's run ! " + Thread.currentThread().getName());
        }
    }
}

File: src/test/kotlin/com/example/sandbox/concurrent/ExecutorTest.kt
package com.example.sandbox.concurrent

import org.junit.jupiter.api.Test
import java.util.concurrent.Executor

class ExecutorTest {

    @Test
    fun test() {
        val executor: Executor = Executor {
            command: Runnable -> command.run()
        }
        executor.execute { println("1") }

    }


    internal class DirectExecutor : Executor {
        override fun execute(r: Runnable) {
            r.run()
        }
    }

}